src/components/dashboard/Claim.js

```diff
@@ -14,23 +14,53 @@ import logger from '../../lib/logger/pino-logger'
 type ClaimProps = DashboardProps & {
   store: Store
 }
+
+type ClaimState = {
+  loading: boolean,
+  nextClaim: string,
+  claimedToday: any
+}
+
 const log = logger.child({ from: 'Claim' })
 
-class Claim extends Component<ClaimProps, {}> {
+class Claim extends Component<ClaimProps, ClaimState> {
   state = {
-    loading: false
+    loading: false,
+    nextClaim: '23:59:59',
+    claimedToday: {
+      people: '',
+      amount: ''
+    }
   }
+
+  interval = null
+
   goodWalletWrapped = wrapper(goodWallet, this.props.store)
+
   async componentDidMount() {
     //if we returned from facerecoginition then the isValid param would be set
     //this happens only on first claim
+    const { entitlement } = this.props.store.get('account')
     const isValid = this.props.screenProps.screenState && this.props.screenProps.screenState.isValid
     if (isValid && (await goodWallet.isCitizen())) {
       this.handleClaim()
     } else if (isValid === false) {
       this.props.screenProps.goToRoot()
     }
+
+    const claimedToday = await this.goodWalletWrapped.getAmountAndQuantityClaimedToday(entitlement)
+    const nextClaimDate = await this.goodWalletWrapped.getNextClaimTime()
```

Comment:
To make it a bit responsible, probably we can group both promises into a `Promise.all`

```javascript
    const goodWallet = this.goodWalletWrapped
	const [claimedToday, nextClaimDate] = await Promise.all([
        goodWallet.getAmountAndQuantityClaimedToday(entitlement),
        goodWallet.getNextClaimTime()
    ])
```